frontend/src/lib/components/DebugNotice.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect, useState } from 'react'
+import { IconClose } from './icons'
+import { LemonButton } from './LemonButton'
+
+export function DebugNotice(): JSX.Element {
+    const [debugInfo, setDebugInfo] = useState<{ branch: string; revision: string } | undefined>()
+    const [noticeHidden, setNoticeHidden] = useState(false)
+
+    useEffect(() => {
+        const bottomNotice = document.getElementById('bottom-notice')
+        const bottomNoticeRevision = document.getElementById('bottom-notice-revision')?.textContent
+        const bottomNoticeBranch = document.getElementById('bottom-notice-branch')?.textContent
+
+        if (bottomNotice && bottomNoticeRevision && bottomNoticeBranch) {
+            setDebugInfo({
+                branch: bottomNoticeBranch || 'unknown',
+                revision: bottomNoticeRevision || 'unknown',
+            })
+
+            bottomNotice.remove()
+        }
+
+        return () => {}
+    }, [])
+
+    if (!debugInfo || noticeHidden) {
+        return <></>
+    }
+
+    return (
+        <div className="bg-white cursor-pointer border-t" onClick={() => setNoticeHidden(true)}>
+            <div className="py-1 px-2 border-l-4 border-primary text-primary-dark  truncate flex justify-between">
```

Comment:
Subjectively, the close button would be positioned more nicely if `px-2` was instead `pl-2 pr-1`

Reply:
Fair point